### PR TITLE
Up/down buttons

### LINKED ISSUE #153: Item move up/down buttons
Sometimes there is a need for item `[move up]`/`[move down]` buttons. 
For example when item is relatively big comparing to client viewport and D'n'D isn't so easy to do.

I have first dirty attemp ([code](https://github.com/StepicOrg/ember-sortable/compare/master...StepicOrg:item-move-buttons?expand=1)), minimal example:
```handlebars 
{{#sortable-group onChange="reorderItems" as |group|}}
  {{#each model.items as |item|}}
    {{#sortable-item tagName="li" model=item group=group handle=".handle" as |moveUp moveDown|}}
      {{item.name}}
      <span class="handle">&varr;</span>
      <button {{action moveUp}}>&uarr;</button>
      <button {{action moveDown}}>&darr;</button>
    {{/sortable-item}}
  {{/each}}
{{/sortable-group}}
```

Do you need this in upstream? 
What is needed for PR? Tests + docs + demo?

### PR DESCRIPTION
**Pull request by H1D**
Adding actions yielded from `{{sortable-item}}` to move up/down using buttons.
sortable buttons

Closes #153.

 - [x] tests
 - [x] implementation
 - [x] demo
 - [x] docs

---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,22 @@ To change this property, define `spacing` on `sortable-item` (default is `0`):
...
+`{{sortable-item}}` yields hash with two actions: `moveUp` and `moveDown`. Here is how you can use them:
+
+```hbs
+{{#sortable-group onChange="reorderItems" as |group|}}
+  {{#each model.items as |item|}}
+    {{#sortable-item tagName="li" model=item group=group handle=".handle" as |sortItem|}}
+      {{item.name}}
+      <button onclick={{sortItem.moveUp}}>&uarr;</button>
+      <button onclick={{sortItem.moveDown}}>&darr;</button>
+      <span class="handle">&varr;</span>
```
Comment (jgwhite) [inline README.md]: Could you add the missing `{{/sortable-item}}` here?

**Comment by jgwhite** [comment]
Thanks so much for the contribution. We’ll review it soon. In the meantime, could you squash the commits down to a single commit with a nice message such as `Add move{Up,Down} item actions`?

**Comment by H1D** [comment]
@jgwhite maybe I can help somehow?

**Comment by jgwhite** [comment]
Thanks for the nudge @H1D. Could you rebase and squash so we don’t have the Merge commit?

**Comment by H1D** [comment]
@jgwhite done

**Comment by jgwhite** [comment]
Thanks for the nudge.

File: addon/mixins/sortable-item.js [hunk 2]
```diff
@@ -714,5 +729,35 @@ export default Mixin.create({
...
+      }
+      let swapItem = sortedItems[thisIndex + delta];
+      this.set(groupDirection, swapItem.get(groupDirection) + delta);
+    } else {
+      if (thisIndex === (sortedItems.length - 1) ) {
+        return ;
+      }
+      let swapItem = sortedItems[thisIndex + delta];
+      this.set(groupDirection, swapItem.get(groupDirection) + delta);
+    }
```
Comment (jgwhite) [inline addon/mixins/sortable-item.js]: It seems a little indirect to me that we’re dealing in terms of coordinates to perform this operation, when really it’s an operation on an array. Could we somehow skip directly to producing the new array in the order and sending it out with the action?

Comment (H1D) [reply addon/mixins/sortable-item.js]: OK 🤔

Comment (jgwhite) [reply addon/mixins/sortable-item.js]: @H1D by the way — I’m very happy to be persuaded that this is the best approach. Questioning more than pronouncing :)

Comment (H1D) [reply addon/mixins/sortable-item.js]: OK, but I think you are right. This pixel arithmetics looks bad.

Do not see elegant solution here yet. Seems like `{{sortable-group}}` refactoring must be done for that

**Comment by H1D** [comment]
@jgwhite well, I did refactor this.

Now group sorting based on `item.index` which can be set explicitly or via `item.x / item.y` update

**Comment by jgwhite** [comment]
Many thanks @H1D. I’m away over the weekend but will take a look next week.

**Comment by jgwhite** [comment]
Hey, sorry I’ve not gotten back to you on this. I like to take the time to really understand every addition and that time is proving hard to come by! I’ll get there, please keep nudging! Hi

**Comment by H1D** [comment]
👍  OK, I will

**Comment by jgwhite** [comment]
This project is now being maintained by the team at @heroku.

To get the project healthy and maintainable again, we're doing a bit of spring cleaning of older issues and pull requests.

This pull request still seems very valuable and we’d like to resume the conversation around it. Would you mind rebasing it upon master? If you’d rather not, let us know and we’ll look for ways to integrate the ideas.

**Comment by H1D** [comment]
@jgwhite a bit busy this month. But actually I will needs this functionality on my current project in a few month so definitely interested to have this merged.

P.S. I'm curious what happened with idea "let's rewrite this with ember-animated"? https://github.com/heroku/ember-sortable/issues/178

**Comment by jgwhite** [comment]
> P.S. I'm curious what happened with idea "let's rewrite this with ember-animated"?

Haha 😅 I still this might be the future of ember-sortable, but we’d have to wait for ember-animated to become stable and battle-hardened, so it’s more of a distant-future idea.

**Comment by ygongdev** [comment]
Closing this as keyboard has been integrated in `v2`

**Pull request closed without merging**